CoreFoundation/Collections.subproj/CFData.c

```diff
@@ -361,7 +361,10 @@ CF_SWIFT_EXPORT void _CFDataInit(CFMutableDataRef memory, CFOptionFlags flags, C
         }
         
         __CFDataSetNeedsToZero(memory, !cleared);
-        CFDataReplaceBytes(memory, CFRangeMake(0, 0), bytes, length);
+        // Avoid copying from null pointers.
+        if (length > 0) {
+            CFDataReplaceBytes(memory, CFRangeMake(0, 0), bytes, length);
```

Comment:
This is technically a correctness issue (the same portability issue I alluded to above). I figured since it was about knowing which pointers were nullable that it was worth including, but I'll take it out.